.github/workflows/e2e-test-provider-example.yml

```diff
@@ -259,11 +297,137 @@ jobs:
             terraform apply -target module.azure_iam -auto-approve
             terraform apply -target module.azure_infrastructure -auto-approve
             ../build/constellation maa-patch "$(terraform output -raw maa_url)"
-            TF_LOG=INFO terraform apply -target constellation_cluster.azure_example -auto-approve
+            terraform apply -target constellation_cluster.azure_example -auto-approve
           else
-            TF_LOG=INFO terraform apply -auto-approve
+            terraform apply -auto-approve
+          fi
+
+      - name: Cleanup Terraform Cluster on failure
+        # cleanup here already on failure, because the subsequent TF overrides might make the TF config invalid and thus the destroy would fail later
```

Comment:
We should move this to just before the `Update cluster configuration` step to avoid lingering clusters if building the provider fails.
We may still run into problems if updating the cluster configuration itself fails, but I think thats acceptable for now

Reply:
what do you mean? There is no way that this step will fail due to missing Terraform, because Terraform was executed in the step before.

Reply:
Not relevant anymore
When I made the comment there were builds steps after this cleanup which could fail without having a dedicated cleanup of the cluster.